shcomplete/tfdf.py

```diff
@@ -0,0 +1,132 @@
+import os
+import re
+import logging
+import pickle
+import numpy
+import pygtrie
+
+from math import log
+from clint.textui import progress
+
+
+def get_tries(args):
+    """
+    Return a list of tries, one for each file where the keys are the command lines.
+    """
+    log = logging.getLogger("get_tries")
+    tries = []
+    max_length = args.max_length
+    for root, _, files in os.walk(args.directory):
+        for file_name in files:
+            path_to_file = os.path.join(root, file_name)
+            log.info("parsing %s", path_to_file)
+            trie = pygtrie.StringTrie(separator=" ")
+            tries.append((path_to_file, trie))
+            with open(path_to_file, "r") as f:
+                for line in f:
+                    line = [l for l in line.rstrip().replace("\t", " ").split(" ")][:max_length]
+                    key = " ".join(line)
+                    if key not in trie:
+                        trie[key] = 0
+                    node = trie._root
+                    prefixes = []
+                    for i in range(1, len(line)+1):
+                        prefixes.append(" ".join(line[:i]))
+                    for pfx in prefixes:
+                        try:
+                            trie[pfx] += 1
```

Comment:
it's better to use
```
some_dict.get(some_key, some_default_value)
```
so you can make:
```
trie[pfx] = trie.get(pfx, 0) + 1
```
to avoid `try...except`